client/client-lib/src/mint/mod.rs

```diff
@@ -155,6 +153,110 @@ impl MintClient {
         self.context.db.begin_transaction(ModuleRegistry::default())
     }
 
+    /// Adds the final amounts of `change` to the tx before submitting it
+    /// Allows for multiple `change` outputs
+    pub async fn finalize_change(&self, tx: &mut Transaction, change: Vec<Amount>) {
```

Comment:
~Maybe this function should take a mut ref to a DB tx as an argument so that changes can be rolled back if anything fails before the top-level DB tx is committed?~

We currently don't have a DB tx spanning the entire FM tx construction, we probably should. E.g. if something fails in `finalize_change` LN inputs marked as spent will be lost.

Reply:
Yes, that's one thing I realized when I implemented this - we will need to be able to update the db for every module input/output within the same db tx.  I think that is something that could live within `TransactionBuilder`